code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -1,8 +1,145 @@
-/obj/item/flamethrower
+#define FLAMETHROWERMODE 1
+#define WELDERMODE 2
+
+/obj/item/flamethrower //do not spawn
+	name = "flamethrower"
+	desc = "how did you get this?"
+	var/lit = FALSE	//on or off
+	var/operating = FALSE//cooldown
+	var/warned_admins = FALSE //for the message_admins() when lit
+	var/flame_intensity = 0 //for flame intensity
+	var/consumption_rate = 25 // how much fuel is used as a percentage
+	var/tank_exposure = 0 //defensive purposes, namely how likely your to blow up upon being shot holding a flamethrower
+	var/obj/item/tank/plasma/ptank = null
+
+/obj/item/flamethrower/process()
+	var/turf/T = get_turf(src)
+	T.hotspot_expose(700, 2) //start a fire if possible
+	if(prob(5)) //sometimes consume fuel
+		consume_fuel(5) //and not massive chunks of it
+	..()
```

Comment:
For process it's the standard to not call the ..() if this means calling datum/process. datum/process return PROCESS_KILL so if you actually return ..() it will stop processing